pkg/api/pod/util_test.go

```diff
@@ -1187,50 +1187,59 @@ func TestDropProbeGracePeriod(t *testing.T) {
 		},
 	}
 
-	enabled := true
-	for _, oldPodInfo := range podInfo {
-		for _, newPodInfo := range podInfo {
-			oldPodHasGracePeriod, oldPod := oldPodInfo.hasGracePeriod, oldPodInfo.pod()
-			newPodHasGracePeriod, newPod := newPodInfo.hasGracePeriod, newPodInfo.pod()
-			if newPod == nil {
-				continue
-			}
+	//enabled := true
+	for _, enabled := range []bool{true, false} {
+		for _, oldPodInfo := range podInfo {
+			for _, newPodInfo := range podInfo {
 
-			t.Run(fmt.Sprintf("feature enabled=%v, old pod %v, new pod %v", enabled, oldPodInfo.description, newPodInfo.description), func(t *testing.T) {
+				oldPodHasGracePeriod, oldPod := newPodInfo.hasGracePeriod, oldPodInfo.pod()
 
-				var oldPodSpec *api.PodSpec
-				if oldPod != nil {
-					oldPodSpec = &oldPod.Spec
-				}
-				dropDisabledFields(&newPod.Spec, nil, oldPodSpec, nil)
+				newPodHasGracePeriod, newPod := newPodInfo.hasGracePeriod, newPodInfo.pod()
 
-				// old pod should never be changed
-				if !reflect.DeepEqual(oldPod, oldPodInfo.pod()) {
-					t.Errorf("old pod changed: %v", cmp.Diff(oldPod, oldPodInfo.pod()))
+				if newPod == nil {
+					continue
 				}
 
-				switch {
-				case enabled || oldPodHasGracePeriod:
-					// new pod should not be changed if the feature is enabled, or if the old pod had subpaths
-					if !reflect.DeepEqual(newPod, newPodInfo.pod()) {
-						t.Errorf("new pod changed: %v", cmp.Diff(newPod, newPodInfo.pod()))
-					}
-				case newPodHasGracePeriod:
-					// new pod should be changed
-					if reflect.DeepEqual(newPod, newPodInfo.pod()) {
-						t.Errorf("new pod was not changed")
+				t.Run(fmt.Sprintf("feature enabled=%v, old pod %v, new pod %v", enabled, oldPodInfo.description, newPodInfo.description), func(t *testing.T) {
+
+					defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ProbeTerminationGracePeriod, enabled)()
+
+					var oldPodSpec *api.PodSpec
+					if oldPod != nil {
+						oldPodSpec = &oldPod.Spec
 					}
-					// new pod should not have subpaths
-					if !reflect.DeepEqual(newPod, podWithoutProbeGracePeriod()) {
-						t.Errorf("new pod had probe-level terminationGracePeriod: %v", cmp.Diff(newPod, podWithoutProbeGracePeriod()))
+					dropDisabledFields(&newPod.Spec, nil, oldPodSpec, nil)
```

Comment:
`newPod` and `newPodInfo.pod()` should be different after the feature is disabled and the feature is dropped via the `dropDisabledFields` function with the `dropDisabledFields` function written as is right now, right? The reason is that the `dropDisabledFields` function blanks out the `LivenessProbe.TerminationGracePeriodSeconds` and `StartupProbe.TerminationGracePeriodSeconds` fields of the containers in `newPod` while `newPodInfo.pod()` remains unchanged. So `reflect.DeepEqual(newPod, newPodInfo.pod())` should return a `false` for the con

Reply:
What you're discussing here should be covered by the `case newPodHasGracePeriod:` below, no?

Reply:
If I throw an error directly for `case newPodHasGracePeriod`, 5 test cases fail. So I tried adding `if reflect.DeepEqual(newPod, newPodInfo.pod())` under `case newPodHasGracePeriod` and a `t.Errorf` under that `if` condition. The 5 test cases fail for the latter as well.